File: 02-Minimum Swaps to Group All 1's Together II/solution.py
class Solution:
    def minSwaps(self, nums):
        total = sum(nums)
        res = count = sum(nums[:total])

        for i in range(total, len(nums) + total):
            count += nums[i % len(nums)]
            count -= nums[(i - total + len(nums)) % len(nums)]
            res = max(res, count)

        return total - res

File: 02-Minimum Swaps to Group All 1's Together II/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int minSwaps(vector<int>& nums) {
        int total = accumulate(nums.begin(), nums.end(), 0);
        int count = accumulate(nums.begin(), nums.begin() + total, 0);
        int res = count;


        for (int i = total; i < nums.size() + total; ++i) {
            count += nums[i % nums.size()] - nums[(i - total + nums.size()) % nums.size()];
            res = max(res, count);
        }
        return total - res;
    }
};